### ISSUE TITLE
How to join a model from distributed replicas?

### ISSUE THREAD

**Issue by AVancans** [pytorch/xla#1672]
## ❓ Questions and Help

From what I understand, a deepcopy model replica is deployed on each XLA device and then individually trained on non-overlapping data (via ParallelLoader). Each model can then be also evaluated via the same approach.

I would like to save a model for production use. But now I assume I have 8 different models (from each XLA core) trained on different data batches. Are they all synced after training and have the same weights (so I just choose random XLA model) or do I need to somehow manually merge them?

**Comment by jysohn23** [comment]
Actually, at each XLA device takes a forward pass, all the gradients of the model are reduced via cross-replica sum and then the average gradient of all the devices is used to update the gradients. Thus, each XLA device hold identical weights as all other devices. So for production use you should be able to use any and all of the weights.

For saving the weights, refer to our API_GUIDE: https://github.com/pytorch/xla/blob/master/API_GUIDE.md#saving-and-loading-xla-tensors and use `xm.save` API to save your model.
